auth/detect/internal/impersonate/impersonate.go

```diff
@@ -73,6 +76,16 @@ type Options struct {
 	Client *http.Client
 }
 
+func (o *Options) validate() error {
+	if o.Tp == nil {
+		return errors.New("detect: missing required 'source_credentials' field in credentials")
```

Comment:
Should this be labeled "impersonate"? Or even if "detect" is the surface, should we somehow identify the `impersonate.Options` origin of this error?

Reply:
worked in the word impersonated. Should be detect though because that is the public package this error is surfaced through. Either that or we drop the detect, but we have used a similar convention in the past.